Datastructures/queue_template.cpp

```diff
@@ -0,0 +1,104 @@
+#include<iostream>
+using namespace std;
+
+template <class t>
+class MyQueue
+{
+    int s , f , e ;
+    t * arr;
+public:
+    MyQueue()
+    {
+        s=0;
+        f=e=-1;
+    }
+    MyQueue(int n)
+    {
+        s = n;
+        arr = new t[s];
+        f = e = 0;
+    }
+    int Size()
+    {
+        return s;
+    }
+    t Front()
+    {
+        return arr[f];
+    }
+    t Back()
+    {
+        return arr[e];
+    }
+    bool isFull()
+    {
+        if((e + 1) == s)
+            return true;
+        else
+            return false;
+    }
+    bool isEmpty()
+    {
+        if( e == f )
+            return true;
+        else
+            return false;
+    }
+    void Insert(t x)
+    {
+        if(!isFull())
+        {
+            arr[e] = x;
+            e++;
+            s++;

```

Comment:
Your **Queue would never be full** because you are **updating the s** _(size)_ too.